src/batch.rs

```diff
@@ -133,6 +139,18 @@ impl<'a> BatchBuilder<'a> {
                          primitive: Primitive,
                          vertices: &mut [PackedVertex],
                          tile_params: Option<TileParams>) {
+        // FIXME(pcwalton): Awful hack to work around a panic.
+        if self.vertex_buffer.vertices.len() + vertices.len() > u16::MAX as usize {
```

Comment:
Have you got any info on how to repro this? I would be keen to debug this and find the root cause if we have a repro (doesn't need to stop this merging though).